loop_invariants.patch

```diff
@@ -144,27 +144,35 @@ index d8694f0..761c44b 100644
      {
          c = buf[ i ];
          i++;
-@@ -1407,6 +1472,9 @@ static bool objectSearch( const char * buf,
+@@ -1144,6 +1209,7 @@ static JSONStatus_t skipCollection( const char * buf,
+ 
+                     if( skipSpaceAndComma( buf, &i, max ) == true )
+                     {
++                        __CPROVER_assume( isOpenBracket_(stack[depth]));
```

Comment:
I have verified by code inspection that this assumption should hold for the loop - I would ask that reviewers verify this as well - see comment about redundant isOpenBracket_ check.